packages/public/server/src/module/Entity/src/Controller/RepositoryController.php

```diff
@@ -437,24 +438,19 @@ protected function handleAddRevisionPost(
                 'Your revision has been saved and is available'
             );
             $route = 'entity/page';
+            $hash = '#revision-saved-accepted';
         } else {
             $successMessage = $translator->translate(
                 'Your revision has been saved, it will be available once it gets approved'
             );
             $route = 'entity/repository/history';
+            $hash = '#revision-saved';
         }
         $this->getEntityManager()->flush();
         $this->flashMessenger()->addSuccessMessage($successMessage);
 
-        if ($this->getRequest()->isXmlHttpRequest()) {
-            return $this->plugin('url')->fromRoute($route, [
-                'entity' => $entity->getId(),
-            ]);
```

Comment:
You don't seem to handle `XMLHttpRequests` anymore. What's the reason for that? This was necessary for POST requests from JavasScript to the URL (e.g. Edtr.io).

Reply:
Before the function returned the url as a string if it was an XMLHttpRequest and a Zend Redirect Response if not. Now it only returns a string. For the old markdown editor the string is converted into a zend redirect at https://github.com/serlo/serlo.org/blob/c0739c6fb060583f58e5b874281b23df575f443f/packages/public/server/src/module/Entity/src/Controller/RepositoryController.php#L88-L93